client/src/store/mutations/player.ts

```diff
@@ -11,6 +11,10 @@ function SET_PLAYER_ROLE(state: State, payload: Role) {
   state.role = payload;
 }
 
+function UPDATE_PLAYER_ROLE(state: State, payload: {data: Role, role: Role}) {
```

Comment:
hey @Nick42791 what's the purpose of this new mutation?

Reply:
Was it get around the type system constraint that you needed a role mutation? You can get around this with something like

```typescript
type ServerResponse = {
  [key in Exclude<keyof PlayerData, 'role'>]:  keyof typeof Mutations
}
```

https://www.typescriptlang.org/docs/handbook/utility-types.html#excludetu

Reply:
Calvin was exactly right. I'll fix it really quick. Thanks!

Reply:
I'll put up a commit with the logic that removes role. and we can try that out too.